### PR TITLE
Edit from doc view and pdf import

### PR DESCRIPTION
**Pull request by darren2pro**

### CODE REVIEW

File: Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift
Comment (zognin) [inline Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift:19]: Can we remove this? Seems like we shouldn't allow users to go to document view if there is no view model? But seems like you removed it in the other branch?

File: Annotato/Annotato/Views/Document/Edit/DocumentView.swift
Comment (zognin) [inline Annotato/Annotato/Views/Document/Edit/DocumentView.swift:34]: On my screen, the pdf exceeds the bounds on the right and bottom sides, is it possible to fix it?
photo_2022-03-17 10 30 35

File: Annotato/Annotato/Views/Document/List/DocumentListCollectionCellView.swift
Comment (zognin) [inline Annotato/Annotato/Views/Document/List/DocumentListCollectionCellView.swift]: Can we remove the print statement? Or log it as an error?

**Review by zognin — COMMENTED** [review]
LGTM minor changes left

File: Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift
Comment (darren2pro) [reply Annotato/Annotato/Views/Document/Edit/DocumentEditViewController.swift:19]: Yups I removed it in another PR

File: Annotato/Annotato/Views/Document/Edit/DocumentView.swift
Comment (darren2pro) [reply Annotato/Annotato/Views/Document/Edit/DocumentView.swift:34]: Ok I will explore more about this first and resolve it in another PR
I created an issue already

File: Annotato/Annotato/Views/Document/List/DocumentListViewController.swift
Comment (zognin) [inline Annotato/Annotato/Views/Document/List/DocumentListViewController.swift]: I just realised that it shouldn't be the responsibility of goToDocumentEdit to decide how to create a new view model.

can you replace it with

```suggestion
        let selectedDocumentPdfViewModel = DocumentPdfViewModel(baseFileUrl: document.baseFileUrl)
        let documentViewModel = DocumentViewModel(
            annotations: [], // Since there are no annotations at point of upload
            pdfDocument: documentPdfViewModel
        )
        goToDocumentEdit(documentViewModel: documentViewModel)
```

**Review by zognin — COMMENTED** [review]
LGTM, I added a comment, after that is changed it can be merged in

File: Annotato/Annotato/Views/Document/List/DocumentListViewController.swift
Comment (darren2pro) [reply Annotato/Annotato/Views/Document/List/DocumentListViewController.swift]: I think it has to be there because I need access to the view controller to set the currentDocumentViewModel
Otherwise the goToDocumentEdit needs to return the view controller which is not ideal

image

Comment (zognin) [reply Annotato/Annotato/Views/Document/List/DocumentListViewController.swift]: @darren2pro As in instead of passing in `documentPdfViewModel`, you pass in `documentViewModel`. You will still be setting currentDocumentViewModel inside goToDocumentEdit. I'll change it for you and merge this.

Comment (zognin) [reply Annotato/Annotato/Views/Document/List/DocumentListViewController.swift]: Besides it's wrong to create `documentViewModel` based on the pdf inside the navigation. How are you supposed to get the other attributes of `documentViewModel` in this way??

File: Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift
Comment (zognin) [inline Annotato/Annotato/ViewModels/Document/DocumentViewModel.swift:7]: @zognin Note to self: To rearrange attributes in next pr, not doing in this pr to reduce difficulties of merge conflicts.

**Review by zognin — APPROVED** [review]
LGTM, has to be resolved in next PR:

- Rename `currentDocumentViewModel` in `DocumentEditViewController` to `viewModel`
- Remove black rectangle showing pdf boundary
- Make pdf be within boundary
- Rearrange order of attributes in document view models according to https://github.com/Annotato/annotato/pull/24#discussion_r828388581

To be resolved in integration pr

- Add id attribute to `DocumentViewModel` and `DocumentListViewModel`

**Pull request merged**
